auth0/resource_auth0_connection_test.go

```diff
@@ -148,6 +154,10 @@ resource "auth0_connection" "my_connection" {
 	name = "Acceptance-Test-Connection-{{.random}}"
 	is_domain_connection = true
 	strategy = "auth0"
+	metadata = {
+		key1 = "foo"
+		key2 = "bar"
+	}
```

Comment:
If you notice I pushed this in a separate commit, the reason for this is because on the update we were trying to remove the metadata. The problem right now is that if we want to reset this and remove it completely it won't get sent to the API because of https://github.com/auth0/terraform-provider-auth0/issues/14. The Metadata is of type `map[string]string` and has the omitempty tag on it. This is rather unfortunate and is still on our mind to fix soon. Because of this the test was fail constantly showing a diff (the api always was returning metadata as we weren't able to set it to empty). 

We might change this field to be a pointer to a map such that the omitempty won't omit `map[string]string{}` but only `nil`.